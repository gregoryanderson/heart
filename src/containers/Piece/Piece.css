.frame {
  /* border: 3px solid green; */
  box-sizing: border-box;
  position: relative;
  background: black;
  padding: 7px;
  box-shadow: -1px 1px var(--blur) 1px rgba(0, 0, 0, 0.1),
    -2px 2px var(--blur) 1px rgba(0, 0, 0, 0.09),
    -3px 3px var(--blur) 1px rgba(0, 0, 0, 0.08),
    -4px 4px var(--blur) 1px rgba(0, 0, 0, 0.07),
    -5px 5px var(--blur) 1px rgba(0, 0, 0, 0.06),
    -6px 6px var(--blur) 1px rgba(0, 0, 0, 0.05),
    -7px 7px var(--blur) 1px rgba(0, 0, 0, 0.04),
    -8px 8px var(--blur) 1px rgba(0, 0, 0, 0.03),
    -9px 9px var(--blur) 1px rgba(0, 0, 0, 0.03),
    -10px 10px var(--blur) 1px rgba(0, 0, 0, 0.03),
    -11px 11px var(--blur) 1px rgba(0, 0, 0, 0.03),
    -12px 12px var(--blur) 1px rgba(0, 0, 0, 0.02),
    -13px 13px var(--blur) 1px rgba(0, 0, 0, 0.02),
    -14px 14px var(--blur) 1px rgba(0, 0, 0, 0.01),
    -15px 15px var(--blur) 1px rgba(0, 0, 0, 0.01),
    -16px 16px var(--blur) 1px rgba(0, 0, 0, 0.01);
}

.inlay {
  box-shadow: inset -1px 1px 6px 1px rgba(0, 0, 0, 0.24);
  background: whitesmoke;
  align-items: center;
  padding: 75px;
}

p {
  margin: 0;
}

.Piece {
  padding-bottom: 75px;
  display: flex;
  justify-content: space-around;
  --blur: 2px;
}

.bottom-icons {
  display: flex;
  justify-content: space-between;
}

img {
  border: 1px solid black;
}

h3,
h2 {
  margin: 0;
}

.selected {
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
}
