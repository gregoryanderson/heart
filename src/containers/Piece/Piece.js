import "./Piece.css";
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { FaHeart, FaRegHeart, FaQuestion } from "react-icons/fa";
import { Link } from "react-router-dom";

export const Piece = ({
  isFav,
  url,
  width,
  key,
  alt,
  id,
  handleFavorite,
  title,
  artist,
  route
}) => {
  console.log(route)
  return (
    <div className="Piece">
      <div className="frame">
        <section className="inlay">
          {title && <h2>{title}</h2>}
          {artist && <h3>{artist}</h3>}
          {title && (
            <Link to={"/"} className="back-btn">
              â—€ back
            </Link>
          )}
          <img
            className={isFav ? "favorite" : "notFavorite"}
            src={url}
            width={width}
            key={key}
            alt={alt}
            isFav={isFav}
            onClick={() => handleFavorite(id)}
          />
          <div className="bottom-icons">
            <p>{isFav ? <FaHeart /> : <FaRegHeart />}</p>
            <Link to={`/paintings/${id}`}>
              <p>
                <FaQuestion />
              </p>
            </Link>
          </div>
        </section>
      </div>
    </div>
  );
};

export default Piece;
