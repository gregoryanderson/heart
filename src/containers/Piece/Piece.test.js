import React from "react";
import { shallow } from "enzyme";
import { Piece } from "./Piece";

describe("Piece", () => {
  let wrapper, mockHandleFavorite;

  beforeEach(() => {
    mockHandleFavorite = jest.fn();

    wrapper = shallow(
      <Piece
        url={'www.com.com'}
        key= {12345}
        id={12345}
        width="700"
        alt="Dutch Painting"
        isFav={true}
        handleFavorite={mockHandleFavorite}
        route='artist'
      />
    );
  });

  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
  })

  it('should call handleFavorite when clicked', () => {

    // const mockEvent = { preventDefault: jest.fn() }

    wrapper.find('img').simulate('click')

    expect(mockHandleFavorite).toHaveBeenCalled();
  });
});

