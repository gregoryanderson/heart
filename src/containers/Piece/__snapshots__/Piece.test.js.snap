// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piece should match the snapshot 1`] = `
<div
  className="Piece"
>
  <div
    className="frame"
  >
    <section
      className="inlay"
    >
      <img
        alt="Dutch Painting"
        className="favorite"
        isFav={true}
        onClick={[Function]}
        src="www.com.com"
        width="700"
      />
      <div
        className="bottom-icons"
      >
        <p>
          <FaHeart />
        </p>
        <Link
          to="/paintings/12345"
        >
          <p>
            <FaQuestion />
          </p>
        </Link>
      </div>
    </section>
  </div>
</div>
`;
